File: snippets/php/label_a_contact_with_a_tag.php
require_once('/path/to/MailchimpMarketing/vendor/autoload.php');

$mailchimp = new MailchimpMarketing\ApiClient();

$mailchimp->setConfig([
    'apiKey' => 'YOUR_API_KEY',
    'server' => 'YOUR_SERVER_PREFIX'
]);

$email = "urist.mcvankab@example.com";
$list_id = "YOUR_LIST_ID";
$subscriber_hash = md5(strtolower($email));

try {
  $mailchimp->lists->updateListMemberTags($list_id, $subscriber_hash, [
    "tags" => [
      "name" => "Influencer",
      "status" => "active"
    ]
  ]);

    echo "The return type for this endpoint is null";
} catch (MailchimpMarketing\ApiException $e) {
    echo $e->getMessage();
}erTags);
    echo "The return type for this endpoint is null";
} catch (MailchimpMarketing\ApiException $e) {
    echo $e->getMessage();
}

File: snippets/ruby/label_a_contact_with_a_tag.rb
require "MailchimpMarketing"
require "digest"

mailchimp = MailchimpMarketing::Client.new
mailchimp.set_config({
  :api_key => ENV["API_KEY"],
  :server => ENV["API_SERVER"]
})
list_id = ENV["LIST_ID"]

email = "urist.mcvankab@example.com"
subscriber_hash = Digest::MD5.hexdigest email.downcase

response = mailchimp.lists.update_list_member_tags list_id, subscriber_hash, {
  body: {
    tags: [
      {
        name: "Influencer",
        status: "active"
      }
    ]
  }
}

puts "The return type for this endpoint is nil, so this should be true: #{response.nil?}" 